### PR TITLE
CRDCDH-1649 Collaborator can View/Edit Data Submissions

### PR DESCRIPTION
**Pull request by Alejandro-Vega**
### Overview

Restrict certain Data Submission actions for collaborators based on their granted permissions.

### Change Details (Specifics)

- Collaborator "Can View" permission restrictions:
  - Disable "Choose Files" button
  - Disable "Upload" button
  - Disable "Download Configuration File" button (Metadata and Data Files submissions only)
  - Disable all validation options, including validation type and target, as well as the "Validate" button
  - Can view manage collaborators dialog, but cannot modify
  - Disable all Submitter-specific (Collaborators can only be submitters) workflow actions such as "Submit", "Withdraw", and "Cancel"
  - Can't upload Data Files via CLI
  - Can't delete data, also disabled the delete checkboxes in Data View
- Collaborator "Can Edit" permission allows for any and all actions except for modifying collaborators

### Related Ticket(s)

[CRDCDH-1649](https://tracker.nci.nih.gov/browse/CRDCDH-1649) (Edit Task)
[CRDCDH-1690](https://tracker.nci.nih.gov/browse/CRDCDH-1690) (Edit User Story)
[CRDCDH-1738](https://tracker.nci.nih.gov/browse/CRDCDH-1738) (View Task)
[CRDCDH-1639](https://tracker.nci.nih.gov/browse/CRDCDH-1639) (View User Story)

### CODE REVIEW

**Review by amattu2 — CHANGES_REQUESTED** [review]
Two very minor things annotated, everything else LGTM.

File: src/components/DataSubmissions/MetadataUpload.tsx [hunk 1]
```diff
@@ -106,7 +106,11 @@ const MetadataUpload = ({ submission, readOnly, onCreateBatch, onUpload }: Props
   const [isUploading, setIsUploading] = useState<boolean>(false);
   const uploadMetadataInputRef = useRef<HTMLInputElement>(null);
   const isSubmissionOwner = submission?.submitterID === user?._id;
-  const canUpload = UploadRoles.includes(user?.role) || isSubmissionOwner;
+  const collaborator = submission?.collaborators?.find((c) => c.collaboratorID === user?._id);
+  const canUpload =
+    (collaborator && collaborator.permission === "Can Edit") ||
```
[the remark this answers is not in the document]
Comment (Alejandro-Vega) [reply src/components/DataSubmissions/MetadataUpload.tsx]: Thanks for catching this, I did not notice. I modified the logic in [a27d7f1](https://github.com/CBIIT/crdc-datahub-ui/pull/512/commits/a27d7f14911a1bdfe296b14f744a47f2602823d9), LMK if it still needs adjusting.

File: src/components/DataSubmissions/ValidationControls.tsx [hunk 2]
```diff
@@ -229,7 +237,11 @@ const ValidationControls: FC = () => {
         <StyledValidateButton
           variant="contained"
           color="info"
-          disabled={(!canValidateFiles && !canValidateMetadata) || isValidating}
+          disabled={
+            (!canValidateFiles && !canValidateMetadata) ||
+            isValidating ||
+            (collaborator && collaborator.permission !== "Can Edit")
```
[the remark this answers is not in the document]
Comment (Alejandro-Vega) [reply src/components/DataSubmissions/ValidationControls.tsx]: Agreed, thanks! Updated in [059f4af](https://github.com/CBIIT/crdc-datahub-ui/pull/512/commits/059f4af768c562ff86b09ca0b989cb4a85acbc75).

**Review by amattu2 — APPROVED** [review]
LGTM! Didn't retest locally, but the code looks solid.

**Pull request merged**